src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -23,6 +28,102 @@ protected void SetHeadersAndLog(ActionContext context, FileResult result)
             Logger.FileResultExecuting(result.FileDownloadName);
         }
 
+        private long SetContentLength(ActionContext context, RangeItemHeaderValue range)
+        {
+            long start = range.From.Value;
+            long end = range.To.Value;
+            long length = end - start + 1;
+            var response = context.HttpContext.Response;
+            response.ContentLength = length;
+            return length;
+        }
+
+        protected long SetRangeHeaders(ActionContext context, FileResult result, RangeItemHeaderValue range)
+        {
+            var method = context.HttpContext.Request.Method;
+            var isGet = string.Equals("GET", method, StringComparison.OrdinalIgnoreCase);
+            if (!isGet)
+            {
+                result.EnableRangeProcessing = false;
+            }
+
+            if (result.EnableRangeProcessing)
+            {
+                var response = context.HttpContext.Response;
+                response.Headers[HeaderNames.AcceptRanges] = "bytes";
+                long rangeLength = SetContentLength(context, range);
+                return rangeLength;
+            }
+
+            return default(long);
+        }
+
+        protected RangeItemHeaderValue SetContentRangeAndStatusCode(
+            ActionContext context,
+            DateTimeOffset lastModified,
+            EntityTagHeaderValue etag,
+            long fileLength)
+        {
+            var ranges = ParseRange(context, lastModified, etag, fileLength);
+            var response = context.HttpContext.Response;
+            var httpResponseHeaders = response.GetTypedHeaders();
+            bool rangeNotSatisfiable = false;
+            if (ranges == null || ranges.Count != 1)
```

Comment:
This is overkill. It's not enough for ranges to be turned on, you must also do a soft-check for the right range headers and fall back to a normal response.
https://github.com/aspnet/StaticFiles/blob/dev/src/Microsoft.AspNetCore.StaticFiles/StaticFileMiddleware.cs#L103
https://github.com/aspnet/StaticFiles/blob/dev/src/Microsoft.AspNetCore.StaticFiles/StaticFileContext.cs#L221-L234